src/db_adapters/transactions.rs

```diff
@@ -1,22 +1,56 @@
 use actix_diesel::dsl::AsyncRunQueryDsl;
-use diesel::PgConnection;
-use futures::future::join_all;
+use anyhow::Context;
+use diesel::{ExpressionMethods, PgConnection, QueryDsl};
+use futures::future::try_join_all;
+
+use near_indexer::near_primitives;
 
 use crate::models;
 use crate::schema;
 
-/// Saves Transaction to database
+/// Saves Transactions to database
 pub(crate) async fn store_transactions(
     pool: &actix_diesel::Database<PgConnection>,
     shards: &[near_indexer::IndexerShard],
-    block_hash: &str,
+    block_hash: &near_indexer::near_primitives::hash::CryptoHash,
     block_timestamp: u64,
-) {
-    if shards.is_empty() {
-        return;
+    block_height: near_primitives::types::BlockHeight,
+) -> anyhow::Result<()> {
+    let mut tried_to_insert_transactions_count = 0;
+    let tx_futures = shards
+        .iter()
+        .filter_map(|shard| shard.chunk.as_ref())
+        .map(|chunk| {
+            tried_to_insert_transactions_count += chunk.transactions.len();
+            store_chunk_transactions(
+                pool,
+                chunk
+                    .transactions
+                    .iter()
+                    .enumerate()
+                    .collect::<Vec<(usize, &near_indexer::IndexerTransactionWithOutcome)>>(),
+                &chunk.header.chunk_hash,
+                block_hash,
+                block_timestamp,
+                "",
+            )
+        });
+
+    try_join_all(tx_futures).await?;
+
+    let inserted_receipt_ids = collect_converted_to_receipt_ids(pool, block_hash).await?;
+    // If the number is the same, I see no chance if there's something wrong, so we can return here
+    if inserted_receipt_ids.len() == tried_to_insert_transactions_count {
+        return Ok(());
     }
 
-    let futures = shards
+    // https://github.com/near/near-indexer-for-explorer/issues/84
+    // TLDR: it's the hack to store transactions with collided hashes
+    // It should not happen, but unfortunately it did,
+    // we have ~10 such transactions in Mainnet for now
+    let transaction_hash_suffix = "_issue84_".to_owned() + &block_height.to_string();
```

Comment:
It looks more like a prefix

Reply:
We craft a prefix here to pass it into the store_chunk_transactions function below